src/components/Checkbox.tsx

```diff
@@ -0,0 +1,90 @@
+import { useRef } from 'react';
+import { useCheckbox } from 'react-aria';
+import { useToggleState } from 'react-stately';
+import { styled } from '../../stitches.config';
+import { Check16pxIcon } from '../icons';
+
+const Label = styled('label', {
+  '& input:checked:disabled + div': {
+    color: '$gray50',
+  },
+  display: 'block',
+  position: 'relative',
+});
+
+const Input = styled('input', {
+  '&:checked': {
+    backgroundColor: '$brandYellow',
+    borderColor: '$brandYellow',
+  },
+  '&:checked&:disabled': {
+    backgroundColor: '$yellow20',
+    borderColor: '$yellow20',
+  },
+  '&:disabled': {
+    borderColor: '$gray30',
+  },
+  '&:focus': {
+    outline: '2px solid $brandYellow',
+    outlineOffset: '2px',
+  },
+  '&:indeterminate': {
+    borderColor: '$brandYellow',
+    position: 'relative',
+  },
+  '&:indeterminate:after': {
+    backgroundColor: '$brandYellow',
+    content: '""',
+    height: '12px',
+    left: '50%',
+    position: 'absolute',
+    top: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: '12px',
+  },
+  '&[aria-invalid="true"]': {
+    borderColor: '$uiErrorRegular',
+  },
+  appearance: 'none',
+  backgroundColor: '$brandWhite',
+  borderColor: '$gray50',
+  borderRadius: '$4',
+  borderStyle: '$solid',
+  borderWidth: '$thin',
+  height: '24px',
+  margin: '0',
+  position: 'relative',
+  width: '24px',
+});
+
+const IconContainer = styled('div', {
+  left: '50%',
+  position: 'absolute',
+  top: '50%',
+  transform: 'translate(-50%, -50%)',
+});
+
+export const Checkbox = (props: {
+  autoFocus?: boolean;
+  children?: string;
+  defaultSelected?: boolean;
+  isDisabled?: boolean;
+  isIndeterminate?: boolean;
+  validationState?: 'invalid' | 'valid';
```

Comment:
We need a way to control the checkbox as well -- an `onChange` handler that we pass to the input

Reply:
Additional props that are necessary are an `aria-label`, `id` and `name`, `autofocus`, etc.

Now, one great thing about TypeScript is that we can strongly type the all of the native attributes we want to a checkbox using `ComponentPropsWithRef<'input'>` (which you can import from React as a type)

Reply:
Then, you can destructure the bulk of the props like this:

`({ defaultSelected, isDisabled, onClick, ...restProps }: ComponentPropsWithRef<'input'> & { /* additional prop types you want here */ }) => ...` 

and then you can spread this on the input itself, like this:

`<Input onChange={onChange} isDisabled={isDisabled} etc... {...restProps} />` 

You don't need to explicitly destructure anything you don't need to -- and all of the typical native HTML attributes for a checkbox input gets forwarded to `Input`.

Reply:
@Boeing787 I've spent a little time over the weekend looking at this and will need your help next week to understand how this is implemented. I'm too much of a TypeScript noob to workout how this jams with the `react-aria` and `react-stately` setup here.

Reply:
Okay! I've pushed a commit which address these issues - ready for another review 👌